File: slack-poster/src/main/java/dev/number6/slackposter/model/KeyPhrasesAttachment.java
package dev.number6.slackposter.model;

import dev.number6.slackposter.ChannelSummaryImage;

import java.text.MessageFormat;
import java.util.Map;
import java.util.stream.Collectors;

public class KeyPhrasesAttachment extends Attachment {

    public static final String CHANNEL_KEY_PHRASES_DETAILS = "{0} ({1})";

    KeyPhrasesAttachment(ChannelSummaryImage image) {
        this.fallback = "Key Phrases (occurring more than once)";
        this.color = "danger";
        this.title = "Key Phrases (occurring more than once):";

        Map<String, Integer> keyPhrases = image.getKeyPhrasesTotals();
        String keyPhrasesLine = keyPhrases.entrySet().stream()
                .filter(e -> e.getValue() > 1)
                .sorted(Map.Entry.comparingByValue((i1, i2) -> i2 - i1))
                .map(e -> MessageFormat.format(CHANNEL_KEY_PHRASES_DETAILS, e.getKey(), e.getValue()))
                .collect(Collectors.joining(", "));

        this.fields = new Field[]{new Field("", keyPhrasesLine, false)};
    }
}

File: slack-poster/src/main/java/dev/number6/slackposter/model/KeyPhrasesAttachment.kt
package dev.number6.slackposter.model

import dev.number6.slackposter.ChannelSummaryImage
import java.text.MessageFormat
import java.util.Map.Entry.comparingByValue
import java.util.stream.Collectors

class KeyPhrasesAttachment internal constructor(image: ChannelSummaryImage) : Attachment() {
    companion object {
        const val CHANNEL_KEY_PHRASES_DETAILS = "{0} ({1})"
    }

    init {
        fallback = "Key Phrases (occurring more than once)"
        color = "danger"
        title = "Key Phrases (occurring more than once):"
        val keyPhrases = image.keyPhrasesTotals
        val keyPhrasesLine = keyPhrases.entries.stream()
                .filter { e -> e.value > 1 }
                .sorted(comparingByValue { i1, i2 -> i2 - i1 })
                .map { e -> MessageFormat.format(CHANNEL_KEY_PHRASES_DETAILS, e.key, e.value) }
                .collect(Collectors.joining(", "))
        fields = arrayOf(Field("", keyPhrasesLine, false))
    }
}